### PR TITLE
migrate substation creation dialog

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/components/refactor/dialogs/substation-creation/substation-creation-form.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    const substationCountryField = (
+        <CountrySelectionInput
+            name={COUNTRY}
+            label={'Country'}
+            formProps={filledTextField}
+            size={'small'}
+        />
+    );
+
+    const AdditionalProps = (
```
Comment (archendel) [inline src/components/refactor/dialogs/substation-creation/substation-creation-form.js:53]:
```suggestion
    const additionalProps = (
```

File: src/components/refactor/rhf-inputs/country-selection-input.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                require('localized-countries/data/' +
+                    getComputedLanguage(languageLocal).substr(0, 2))
+            );
+        } catch (error) {
+            // fallback to english if no localised list found
+            return englishCountriesModule;
+        }
+    }, [languageLocal, englishCountriesModule]);
+
+    const values = useMemo(
```
Comment (archendel) [inline src/components/refactor/rhf-inputs/country-selection-input.js:34]:
```suggestion
    const options = useMemo(
```

File: src/components/refactor/dialogs/substation-creation/property/property-form.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import { gridItem } from '../../../../dialogs/dialogUtils';
+import React, { useEffect, useMemo, useState } from 'react';
+import AutocompleteInput from '../../../rhf-inputs/autocomplete-input';
+import { NAME, VALUE } from '../../../utils/field-constants';
+import { fetchPredefinedProperties } from './property-utils';
+import { useWatch } from 'react-hook-form';
+
+const PropertyForm = ({ name, index }) => {
+    const [predefinedProperties, setPredefinedProperties] = useState();
+    const propertyName = useWatch({ name: `${name}.${index}.${NAME}` });
```
Comment (archendel) [inline src/components/refactor/dialogs/substation-creation/property/property-form.js:17]:
```suggestion
    const watchPropertyName = useWatch({ name: `${name}.${index}.${NAME}` });
```

**Review by ayolab — APPROVED** [review]
Tests OK

**Review by archendel — APPROVED** [review]

**Pull request merged**